libsql/src/local/connection.rs

```diff
@@ -57,13 +57,23 @@ impl Connection {
                 )));
             }
         }
-
-        Ok(Connection {
+        let conn = Connection {
             raw,
             drop_ref: Arc::new(()),
             #[cfg(feature = "replication")]
             writer: db.writer()?,
-        })
+        };
+        #[cfg(feature = "sync")]
+        if let Some(_) = db.sync_ctx {
+            // We need to make sure database is in WAL mode with checkpointing
+            // disabled so that we can sync our changes back to a remote
+            // server.
+            conn.query("PRAGMA journal_mode = WAL", Params::None)?;
+            unsafe {
+                ffi::libsql_wal_disable_checkpoint(conn.raw);
```

Comment:
Do we need to adding parsing to prevent users from check pointing on their own?

Reply:
Added a note of that in https://gist.github.com/penberg/c8082b450fa8ba56a96c2d9c0d45ad8f